CRM/Member/BAO/Membership.php

```diff
@@ -105,12 +105,15 @@ public static function add(&$params, $ids = []) {
       'max_related' => $membership->max_related,
     ];
 
-    $session = CRM_Core_Session::singleton();
+    if (!empty($params['modified_id'])) {
+      $membershipLog['modified_id'] = $params['modified_id'];
+    }
     // If we have an authenticated session, set modified_id to that user's contact_id, else set to membership.contact_id
-    if ($session->get('userID')) {
-      $membershipLog['modified_id'] = $session->get('userID');
+    elseif (CRM_Core_Session::singleton()->get('userID')) {
+      $membershipLog['modified_id'] = CRM_Core_Session::singleton()->get('userID');
     }
     elseif (!empty($ids['userId'])) {
```

Comment:
@eileenmcnaughton just wondering if we could do something like this earlier up

```php
if (empty($params['modified_id'] && !empty($ids['userId'])) {
  $params['modified_id'] = $ids['userId'];
}
```

Reply:
@seamuslee001 so next step is to stop  passing $ids into add at all & pass modified_id in from create

Reply:
@eileenmcnaughton We should replace `get('userID')` with `getLoggedInContactID()`

Reply:
@mattwire yes - there is a bunch of follow on tidy up we should do - goal one is just to get $ids out of the params for  the add  function

Reply:
ie once  there is an alternative we can eliminate $ids['userID']